Pipeline/Preprocessing/PreprocessingModule.py

```diff
@@ -114,31 +121,55 @@ def __download_from_youtube(link, path_to_video):
     @staticmethod
     def __preprocess_meta(path_to_meta):
         """
-        Converts meta-info into a unified binary format,
+        Converts meta-info into a unified binary mask format,
         where for each second "1" - "song", "0" - "not song".
+        Converts given name of genre for each segment to genre-mask.
         Extracts YouTube link if given in the 1st line of the meta-info file.
         :param path_to_meta: path to meta-info about the audio;
-        :return: binary mask generated based on given meta and link to YouTube src if given.
+        :return: binary mask (song/not song), genres mask both
+        generated based on given meta and link to YouTube src if given.
         """
         assert os.path.isfile(path_to_meta), "Meta-info file {} not found".format(path_to_meta)
-
+        genres_dict = {
+            'none': 0,  # this label for non-musical segments
+            'pop': 1,
+            'rock': 2,
+            'jazz': 3,
+            'reggae': 4,
+            'metal': 5,
+            'country': 6,
+            'indi': 7,
+            'electronic': 8,
+            'hiphop': 9,
+            'disco': 10,
+            'blues': 11,
+            'folk': 12,
+            'classical': 13
+        }
         bin_mask = []
+        genre_mask = []
         link = None
         with open(path_to_meta) as meta:
             for line in meta:
+                line = line.strip(" \n\t").replace(' ', '')
                 if line.startswith("http"):
-                    link = line.rstrip("\n")
+                    link = line
```

Comment:
for what?

Reply:
@OlegBEZb i don't quite understand what you mean. the 1st line of the file is the link to the YT video, this is where it is extracted.

Reply:
oh, I'm sorry. I've read as `line = line`
my oversight(